### DISCUSSION TITLE
Plugin: Load Postgres view into netbox view for fill a table.

### DISCUSSION THREAD

**Discussion by Brescou** [netbox-community/netbox#16401] [Help Wanted!]
Hey everyone,

I'm working on a NetBox plugin and need some help. I want to load data from a PostgreSQL view into a NetBox view to populate a table. Here's what I'm trying to do:

What I'm trying to achieve:
I want a table in a NetBox view to show data from a PostgreSQL view. This data should be fetched dynamically and displayed within NetBox.

Setup:

NetBox version: 4.0.3-dev
PostgreSQL version: 16
Plugin: Custom plugin in development
Questions:

What's the best way to connect a PostgreSQL view to a NetBox view?
Are there any examples or docs that show how to do this?
How can I make sure the data is dynamically fetched and refreshed in the NetBox view?
Any potential issues or things to watch out for?
Extra info:
I know a bit about Django and the NetBox plugin setup but I'm pretty new to PostgreSQL views. Any code examples, links, or tips would be awesome.

Thanks a lot!

**Comment by jeremystretch** [comment]
AFAIK there's no native support in Django for PostgreSQL views, unfortunately. You might look into the [django-pgviews](https://pypi.org/project/django-pgviews/) package for this; I haven't tried it myself. A simpler solution would be to create the view(s) manually and use [the ORM's `raw()` method](https://docs.djangoproject.com/en/5.0/topics/db/sql/) to query. Hope that helps!

**Reply by Brescou** [reply]
Hi Jeremy,

Thank you for your quick response!

I wanted to give you a bit more context on what I'm trying to achieve. I chose to use a PostgreSQL view because I couldn't manage to merge the IPAddress, Device, VirtualMachine, and Impact models (the model I created for my plugin) into a single table. My goal is to create a table in NetBox where the Impact model is declared in the Meta of the table, while displaying data from the other models.

Do you have any specific recommendations to help me overcome these challenges? Any suggestions or resources would be greatly appreciated.

Thanks again for your help!
